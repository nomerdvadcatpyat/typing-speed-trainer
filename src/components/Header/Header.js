import React from 'react';
import {Link} from "react-router-dom";
import './Header.scss';


export function Header() {
	return (
		<header className="header">
			<nav>
				<Link to="/rating"> Рейтинг </Link>
				<Link to="/training"> Тренировка </Link>
				<Link to="/login"> Вход </Link>
			</nav>
		</header>
	);
}