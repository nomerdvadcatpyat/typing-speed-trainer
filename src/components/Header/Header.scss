@import "src/utils/scss-variables/variables";


.header-buffer {
  height: $header-height
}

.header {
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 100;

  display: flex;
  min-height: $header-height;
  font-size: 1.5rem;

  background-color: $primary;

  &__logo {
    position: absolute;
    left: 1rem;
  }
}

@media screen and (max-width: map-get($grid-breakpoints, 'sm')) {
  .header {
    padding-top: 1.5rem;
    flex-flow: column nowrap;

    &__nav {
      margin-top: 3.5rem;
      overflow: hidden;
      display: flex;
      flex-flow: column nowrap;
      text-align: center;
      height: 0;


      &_collapsed {
        animation: close-menu .2s linear forwards;
      }

      &_opened {
        animation: open-menu .2s linear forwards;
      }
    }
  }

  @keyframes open-menu {
    from {
      opacity: 0;
      height: 0;
    }
    1% {
      opacity: 1;
    }
    50% {
      opacity: 1;
      height: 7rem;
    }
    to {
      opacity: 1;
      height: 14rem;
    }
  }

  @keyframes close-menu {
    from {
      height: 14rem;
      opacity: 1;
    }
    50% {
      opacity: 1;
      height: 7rem;
    }
    99% {
      opacity: 1;
    }
    to {
      height: 0;
      opacity: 0;
    }
  }

  .header-link {
    margin-bottom: 1rem;
  }
}

@media screen and (min-width: map-get($grid-breakpoints, 'sm')) {
  .header {
    flex-flow: row nowrap;
    align-items: center;
    justify-content: flex-end;

    &__nav {
      width: 30rem;
      display: flex;
      flex-flow: row nowrap;
      justify-content: space-around;
    }
  }
}