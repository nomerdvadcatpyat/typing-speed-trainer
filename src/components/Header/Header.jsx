import React from 'react';
import {Link} from "react-router-dom";
import './Header.scss';
import PropTypes from "prop-types";

export function Header({ isAuth, onLogout }) {
	return (
		<header className="header">
			<nav>
				<Link to="/rating"> Рейтинг </Link>
				<Link to="/training"> Тренировка </Link>
				{ isAuth? <span className="header__logout-button" onClick={onLogout}> Выход </span> :
					<Link className="header__login-button" to="/auth/login"> Вход </Link> }
			</nav>
		</header>
	);
}

Header.propTypes = {
	onLogout: PropTypes.func,
	isAuth: PropTypes.bool
}
