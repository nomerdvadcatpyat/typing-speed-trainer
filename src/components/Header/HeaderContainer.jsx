import React from 'react';
import {connect} from 'react-redux';
import {Header} from "./Header";
import {bindActionCreators} from "redux";
import {logout} from "../../store/actions/auth";

function mapStateToProps(state) {
	return {
		isAuth: state.user.isAuth
	};
}

function mapDispatchToProps(dispatch) {
	return {
		logout: bindActionCreators(logout, dispatch)
	};
}

const HeaderContainer = (props) => {
	return (
		<Header {...props} />
	);
}

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(HeaderContainer);