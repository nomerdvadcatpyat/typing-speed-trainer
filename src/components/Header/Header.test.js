import { shallow, mount } from 'enzyme';
import {Header} from "./Header";

describe('<Header />', () => {

	it('mount with auth', () => {
		const header = shallow(<Header isAuth={true} />);

		expect(header).toMatchSnapshot();
	});


	it('mount without auth', () => {
		const header = shallow(<Header isAuth={false} />);

		expect(header).toMatchSnapshot();
	});


	it('click on logout', () => {
		const logoutFn = jest.fn();
		const header = shallow(<Header isAuth={true} onLogout={logoutFn}/>);
		header.find('.header__logout-button').simulate('click');

		expect(logoutFn).toHaveBeenCalledTimes(1);
	});

});
