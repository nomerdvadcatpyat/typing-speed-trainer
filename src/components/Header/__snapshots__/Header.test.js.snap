// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> mount with auth 1`] = `
<header
  className="header"
>
  <nav>
    <Link
      to="/rating"
    >
       Рейтинг 
    </Link>
    <Link
      to="/training"
    >
       Тренировка 
    </Link>
    <span
      className="header__logout-button"
    >
       Выход 
    </span>
  </nav>
</header>
`;

exports[`<Header /> mount without auth 1`] = `
<header
  className="header"
>
  <nav>
    <Link
      to="/rating"
    >
       Рейтинг 
    </Link>
    <Link
      to="/training"
    >
       Тренировка 
    </Link>
    <Link
      className="header__login-button"
      to="/auth/login"
    >
       Вход 
    </Link>
  </nav>
</header>
`;
